### PR TITLE
Tech/224/scenario as update object

### LINKED ISSUE #224: Scenarios should only be about metrics
# Feature request
As an auditor, I want scenarios that help me select important metric combinations in one click, so I can quickly switch between different views.

Currently the scenarios reset everything, including weird things such as margin. That is confusing and does not reflect what I want. 

To be usable the scenarios should only be about metrics. The coverage scenario, for example, should only set the three metrics and the color range. A building with >= 80% coverage should be green. A building with >= 20% coverage should be yellow. All other buildings red.

### PR DESCRIPTION
**Pull request by alschmut**
# Tech/224/scenario as update object

closes #224 
Merge permission: @mrpatpat, @alschmut 

## Description

- Scenarios will only have to hold necessary settings, as it will no longer update all settings, but just the settings, which are mentioned inside the scenario
- Tech: That also means, when adding an attribute to `settingsService.settings`, we no longer have to update the (Default)-Scenario object (which is loaded initially) to not loose the value of the new settings-attribute.
- Different from the issue #224: Scenarios are still able to override other settings attributes and not *only metrics*. That makes sense for example for `neutralColorRange`, when the metric `coverage` is used and the `Invert Colors` actually makes sense. Or using predefined range-boundaries for the percentage of coverage.

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

* [x] **All** requirements mentioned in the issue are implemented
* [x] Does match the Code of Conduct and the Contribution file 
* [x] Task has its own **GitHub issue** (something it is solving)
  * [x] Issue number is **included in the commit messages** for traceability
* [x] **Update the README.md** with any changes/additions made
* [x] **Update the CHANGELOG.md** with any changes/additions made
* [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
* [x] **All tests pass**    
* [x] **Descriptive pull request text**, answering:
  + What problem/issue are you fixing?
  + What does this PR implement and how? 
* [ ] **Assign your PR to someone** for a code review
  + This person _will be contacted **first**_ if a bug is introduced into `master`
* [x] **Manual testing** did not fail

### CODE REVIEW

File: visualization/app/codeCharta/core/scenario/scenario.service.ts [hunk 1]
```diff
@@ -52,10 +49,23 @@ export class ScenarioService {
         }
     }
 
-    /**
-     * Returns an array of all scenarios.
-     * @returns {Scenario[]} all scenarios
-     */
+    private updateSettingsUsingScenario(settings, scenarioSettings): any {
```
Comment (mrpatpat) [inline visualization/app/codeCharta/core/scenario/scenario.service.ts]: try using spread operator

File: visualization/app/codeCharta/core/scenario/scenario.data.ts [hunk 2]
```diff
@@ -3,41 +3,21 @@ import {CodeMap} from "../data/model/CodeMap";
...
-            invertHeight: false,
-            dynamicMargin: true,
-            isWhiteBackground: false,
-            blacklist:[],
-            searchPattern: ""
+            colorMetric: "mcc"
         },
         autoFitCamera: true
-    };
+    } as Scenario;
```
Comment (mrpatpat) [inline visualization/app/codeCharta/core/scenario/scenario.data.ts]: use Partial type for settings

File: visualization/app/codeCharta/core/scenario/scenario.service.ts (hunk 1 above)
Comment (alschmut) [reply visualization/app/codeCharta/core/scenario/scenario.service.ts]: Works perfectly. It even works in nested levels 👍

File: visualization/app/codeCharta/core/scenario/scenario.data.ts (hunk 2 above)
Comment (alschmut) [reply visualization/app/codeCharta/core/scenario/scenario.data.ts]: Implemented: see further commits.

**Pull request merged**